cli/config.py

```diff
@@ -224,6 +223,17 @@ def merge(self, uid, username, size, size_low_pri, vm_size, subnet_id, password,
             raise aztk.error.AztkError(
                 "Please supply a value for wait in either the cluster.yaml configuration file or with a parameter (--wait or --no-wait)")
 
+        if self.size > 0 and self.size_low_pri > 0:
+            self.mixed_mode = True
+
+        if not self.subnet_id and self.mixed_mode:
+            raise aztk.error.AztkError(
+                "You must configure a VNET to use AZTK in mixed mode (dedicated and low priority nodes). Set the VNET's subnet_id in your cluster.yaml.")
+
+        # ensure spark_client is built with AAD if using mixed mode
+        if not spark_client.secrets_config.service_principal.tenant_id and self.mixed_mode:
```

Comment:
Minor, since AAD is a requirement for subnets, then I would raise this error first.